dask_geopandas/morton_distance.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+import pandas as pd
+from dask_geopandas.hilbert_distance import _continuous_to_discrete_coords
+
+# Based on #https://code.activestate.com/recipes/
+# 577558-interleave-bits-aka-morton-ize-aka-z-order-curve/
+
+
+def _morton_distance(gdf, total_bounds, p):
+    """
+    Calculate
+
+    Parameters
+    ----------
+
+    gdf : GeoDataFrame
+
+    total_bounds : Total bounds
+
+    p : number of iterations
+
+    """
+    bounds = gdf.bounds.to_numpy()
+    coords = _continuous_to_discrete_coords(total_bounds, bounds, p)
```

Comment:
The `p` parameter is the same as the Hilbert curve has? I mean, denoting the same concept?

Reply:
I believe the following logic holds true - correct me otherwise;
Using the `_continuous_to_discrete_coords` transforms each of the coordinate pairs from continuous to integers.
The p (or order of curve) value will control the number of iterations in constructing the [Hilbert/Morton] curve. 
Then calculated distances along each curve work entirely in terms of integers, where the size of the hypercube will reflect the number of bits per dimension - controlled by p;

* The 1st order curve supports values (0|1) on both x and y axis - giving us 2 bit binary numbers of the range from 0,1,2,3.
* Then the 2nd order curve supports values (0|1|2|3) on ... - giving us 4 bit binary numbers of the range 0,1...,15 and so on...

Therefore, using the same p value (or n iterations) for calculated distances along both curves will be bound by the same size hypercube.
 
The maximum discrete distances along each order are as follows;
```python
iterations = 16 # or p 
n = 2 # two-dimension
max_discrete_distances = []
for p in range(iterations):
    max_discrete_distances.append(2 ** (n*p)  - 1 ) # -1 to start index at 0
max_discrete_distances[1:]  # from 1st to 15th order curve
```

long story short, yes it denotes the same concept.

Reply:
Cool! Thanks for the great explanation!

Reply:
> The p (or order of curve) value will control the number of iterations in constructing the [Hilbert/Morton] curve.

But as far as I can see, to calculate the Morton curve, we are not using any iterations? (in contrast to Hilbert curve) 
The `encode_morton` and `part1by1` seem to do a fixed set of computations independent of `p`.

(the explanation that it controls the size of the hypercube (and thus the precision of the calculated distances) is of course still valid nonetheless)

Reply:
(I suppose the "classical" implementation also uses loops, but the bit-shifting method is an optimized version that doesn't require loops)